internal/storage/db/get.go

```diff
@@ -0,0 +1,188 @@
+				FROM "order" o
+				LEFT JOIN users u ON u.id = o.user_id
+				LEFT JOIN status s ON s.id = o.status_id
+				WHERE u.login = $1`,
+		login)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var accrual float32
+	var number, status, uploadedAt string
+	for rows.Next() {
+		err = rows.Scan(&number, &accrual, &status, &uploadedAt)
+		if err != nil {
+			logger.WriteErrorLog(err.Error())
+			continue
+		}
+		res = append(res, user.Order{
+			Number:     number,
+			Status:     status,
+			Accrual:    accrual,
+			UploadedAt: uploadedAt,
+		})
+	}
+
+	err = rows.Err()
+	if err != nil {
+		logger.WriteErrorLog(err.Error())
+	}
+	return res, err
+}
+
+// GetAllOrdersInStatuses получить все заказы в статусах
+func (s *Storage) GetAllOrdersInStatuses(statuses []int) ([]user.OrderCheckAccrual, error) {
+	var res []user.OrderCheckAccrual
+
+	rows, err := repository.DBRepository.QueryContext(
+		context.Background(),
+		`SELECT number, accrual::DOUBLE PRECISION, s.alias
+				FROM "order" o
+				LEFT JOIN status s ON s.id = o.status_id
+				WHERE status_id = ANY($1)
+				AND (check_accrual_after IS NULL OR check_accrual_after <= $2)
+				LIMIT 50`,
+		pq.Array(statuses),
+		time.Now().Unix())
+
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var accrual float32
+	var number, status string
+	for rows.Next() {
+		err = rows.Scan(&number, &accrual, &status)
+		if err != nil {
+			logger.WriteErrorLog(err.Error())
+			continue
+		}
+		res = append(res, user.OrderCheckAccrual{
+			Number:  number,
+			Accrual: accrual,
+			Status:  status,
+		})
+	}
+
+	err = rows.Err()
+	if err != nil {
+		logger.WriteErrorLog(err.Error())
+	}
+	return res, err
+}
+
+// GetBalance получить баланс пользователя
+func (s *Storage) GetBalance(login string) (balance.Balance, error) {
+	var b balance.Balance
+	row := repository.DBRepository.QueryRowContext(
+		context.Background(),
+		`SELECT
+					"value"::DOUBLE PRECISION AS balance,
+					COALESCE(SUM(w.sum) OVER(PARTITION BY b.id), 0::DOUBLE PRECISION) AS sum
+				FROM balance b
+						 INNER JOIN users u ON u.id = b.user_id
+						 LEFT JOIN withdraw w ON w.user_id = b.user_id
+				WHERE u.login = $1
+				LIMIT 1`,
+		login)
+	err := row.Scan(&b.Current, &b.Withdrawn)
+	return b, err
+}
+
+// GetWithdrawals получение всех списаний баллов
+func (s *Storage) GetWithdrawals(login string) ([]balance.Withdraw, error) {
+	var res []balance.Withdraw
+	rows, err := repository.DBRepository.QueryContext(
+		context.Background(),
+		`SELECT "order", "sum", processed_at
+				FROM withdraw w
+				INNER JOIN users u ON u.id = w.user_id
+				WHERE u.login = $1
+				ORDER BY processed_at DESC`,
+		login)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var sum float32
+	var number, processedAt string
+	for rows.Next() {
+		err = rows.Scan(&number, &sum, &processedAt)
+		if err != nil {
+			logger.WriteErrorLog(err.Error())
+			continue
```

Comment:
```suggestion
			return nil, err
```
Запрос с неконсистентными данными - это хуже, чем просто ошибка при осуществлении запроса. Поэтому никаких continue.
https://t.me/bushigo/40

Reply:
поправил